docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,374 @@
+	}
+
+	// Connect to the oracle service (default timeout of 5 seconds).
+	go func() {
+		if err := oracleClient.Start(context.Background()); err != nil {
+			app.Logger().Error("failed to start oracle client", "err", err)
+			panic(err)
+		}
+
+		app.Logger().Info("started oracle client", "address", cfg.OracleAddress)
+	}()
+	return oracleClient, oracleMetrics, nil
+}
+```
+## ABCI and Vote Extensions
+Configure the ABCI methods and vote extensions.
+
+Define a method to contain the logic where these will be configured.
+```go oracle.go
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {}
+```
+
+Within this method, do the following:
+- **Setup Proposal Handler:** This handler will be used in `PrepareProposal` and `ProcessProposal` to fill proposals with the oracle data.
+- **Set PreBlocker:** The Application's `PreBlocker` will be configured to write price data to state before transactions are executed.
+- **Set Vote Extensions:** Set the vote extension handlers on the Application that will handle adding price data to the node's consensus votes.
+
+Start with setting up the proposal handler.
+
+```go oracle.go
+package app
+
+import (
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+
+    "github.com/cosmos/cosmos-sdk/baseapp"
+    "github.com/skip-mev/slinky/abci/proposals"
+    compression "github.com/skip-mev/slinky/abci/strategies/codec"
+    "github.com/skip-mev/slinky/abci/strategies/currencypair"
+    "github.com/skip-mev/slinky/abci/ve"
+)
+
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {
+    // Create the proposal handler that will be used to fill proposals with
+	// transactions and oracle data.
+	proposalHandler := proposals.NewProposalHandler(
+		app.Logger(),
+		baseapp.NoOpPrepareProposal(),
+		baseapp.NoOpProcessProposal(),
+		ve.NewDefaultValidateVoteExtensionsFn(app.StakingKeeper),
+		compression.NewCompressionVoteExtensionCodec(
+			compression.NewDefaultVoteExtensionCodec(),
+			compression.NewZLibCompressor(),
+		),
+		compression.NewCompressionExtendedCommitCodec(
+			compression.NewDefaultExtendedCommitCodec(),
+			compression.NewZStdCompressor(),
+		),
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+		oracleMetrics,
+	)
+	app.SetPrepareProposal(proposalHandler.PrepareProposalHandler())
+	app.SetProcessProposal(proposalHandler.ProcessProposalHandler())
+}
+```
+
+Next, set up the `PreBlocker`. This involves:
+
+- **Aggregate Function:** Setting the aggregator function that combines all reported prices into one final price per currency pair.
+- **Currency Pair Strategy:** Setting the currency pair strategy. For this example, we will use the `DeltaCurrencyPairStrategy` which encodes/decodes the price as the difference between the current price and the previous price.
```

Comment:
I think we should note that this is the preferred price encoding method as well, unless you have some other specific use case.

Reply:
added a note that this is the recommended strategy